src/OpenTelemetry.Contrib.Instrumentation.AWSLambda/Implementation/AWSLambdaWrapper.cs

```diff
@@ -27,162 +27,216 @@ namespace OpenTelemetry.Contrib.Instrumentation.AWSLambda.Implementation
+        private static readonly ActivitySource AWSLambdaActivitySource = new(AWSLambdaUtils.ActivitySourceName);
 
         /// <summary>
-        /// Tracing wrapper for Lambda handler without Lambda context.
+        /// Gets or sets a value indicating whether AWS X-Ray propagation should be ignored. Default value is false.
+        /// The flag was introduced as a workaround of this issue:
+        /// the ActivitySource.StartActivity method returns null if sampling decision of DROP (Sampled=0).
+        /// Flag can be removed as soon as the bug https://github.com/open-telemetry/opentelemetry-dotnet/issues/3290 is resolved.
         /// </summary>
-        /// <typeparam name="TInput">Input.</typeparam>
-        /// <typeparam name="TResult">Output result.</typeparam>
-        /// <param name="tracerProvider">TracerProvider passed in.</param>
-        /// <param name="lambdaHandler">Lambda handler function passed in.</param>
-        /// <param name="input">Instance of input.</param>
-        /// <returns>Instance of output result.</returns>
-        public static TResult Trace<TInput, TResult>(TracerProvider tracerProvider, Func<TInput, TResult> lambdaHandler, TInput input)
-        {
-            return Intercept(tracerProvider, () => lambdaHandler(input));
-        }
+        internal static bool IgnoreAWSXRayPropagation { get; set; }
 
         /// <summary>
-        /// Tracing wrapper for Lambda handler without Lambda context.
+        /// Tracing wrapper for Lambda handler.
         /// </summary>
         /// <typeparam name="TInput">Input.</typeparam>
+        /// <typeparam name="TResult">Output result.</typeparam>
         /// <param name="tracerProvider">TracerProvider passed in.</param>
         /// <param name="lambdaHandler">Lambda handler function passed in.</param>
         /// <param name="input">Instance of input.</param>
-        public static void Trace<TInput>(TracerProvider tracerProvider, Action<TInput> lambdaHandler, TInput input)
+        /// <param name="context">Instance of lambda context.</param>
+        /// <param name="parentContext">
+        /// The optional parent context <see cref="ActivityContext"/> is used for Activity object creation.
+        /// If no parent context provided, incoming request is used to extract one.
+        /// If parent is not extracted from incoming request then X-Ray propagation is used to extract one.
+        /// </param>
+        /// <returns>Instance of output result.</returns>
+        public static TResult Trace<TInput, TResult>(
+            TracerProvider tracerProvider,
+            Func<TInput, ILambdaContext, TResult> lambdaHandler,
+            TInput input,
+            ILambdaContext context,
+            ActivityContext parentContext = default)
         {
-            Intercept(tracerProvider, () => lambdaHandler(input));
+            TResult result = default;
+            Action action = () => result = lambdaHandler(input, context);
```

Comment:
I would inline the action into the TraceInternal call.

Reply:
I wouldn't do it as it makes code less clear.

Reply:
I think the name `action` doesn't make anything more clear. But OK for me to leave it as-is.

Reply:
Not the name `action` itself, but splitting call into 2 lines instead of in-lining.